services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/statistics/StatisticsEventsCollector.java

```diff
@@ -0,0 +1,31 @@
+package org.sagebionetworks.repo.manager.statistics;
+
+import java.util.List;
+
+import org.sagebionetworks.repo.manager.statistics.events.StatisticsEvent;
+
+/**
+ * Main entry point to collect statistics about {@link StatisticsEvent}s.
+ * 
+ * @author Marco
+ *
+ */
+public interface StatisticsEventsCollector {
+
+	/**
+	 * Accepts the given {@link StatisticsEvent} in order to collect statistics about the event. If this call is invoked
+	 * within a transaction if will be called only after the transaction is committed.
+	 * 
+	 * @param event The event to collect statistics about
+	 */
+	<E extends StatisticsEvent> void collectEvent(E event);
+
+	/**
+	 * Accepts the given list of {@link StatisticsEvent}s in order to collect statistics about the batch of events. If
+	 * this call is invoked within a transaction if will be called only after the transaction is committed.
+	 * 
+	 * @param events A batch of events to collect statistics about
+	 */
+	<E extends StatisticsEvent> void collectEvents(List<E> events);

```

Comment:
Think about having a queue to batch and a background job to push the events to firehose